src/main/java/com/zendesk/maxwell/replay/MaxwellReplayFile.java

```diff
@@ -0,0 +1,378 @@
+
+	private final AbstractProducer producer;
+
+	private final TableCache tableCache;
+
+	private final SchemaStore schemaStore;
+	private final Schema schema;
+
+	private Position lastHeartbeatPosition;
+
+	private long rowCount;
+
+
+	public static void main(String[] args) {
+		try {
+			new MaxwellReplayFile(args).start();
+		} catch (Exception e) {
+			LOGGER.error(e.getMessage(), e);
+		}
+	}
+
+	private MaxwellReplayFile(String[] args) throws SQLException, URISyntaxException, IOException, SchemaStoreException {
+		this.config = new ReplayConfig(args);
+		MaxwellContext context = new MaxwellContext(config);
+
+		Logging.setLevel(config.log_level == null ? DEFAULT_LOG_LEVEL : config.log_level);
+
+		this.producer = context.getProducer();
+		this.tableCache = new TableCache(config.databaseName);
+		this.schemaStore = new ReplayBinlogStore(context.getSchemaConnectionPool(), CaseSensitivity.CONVERT_ON_COMPARE, config);
+		this.schema = schemaStore.getSchema();
+	}
+
+	public void start() {
+		try {
+			config.binlogFiles.forEach(this::replayBinlog);
+		} catch (Exception e) {
+			LOGGER.error(e.getMessage(), e);
+		} finally {
+			daemon();
+		}
+	}
+
+	/**
+	 * Wait producer for replay
+	 */
+	private void daemon() {
+		while (!producer.flushAndClose()) {
+			LOGGER.debug("waiting produce...");
+			LockSupport.parkNanos(1000_000_000L);
+		}
+		LOGGER.info("complete replay: {}", rowCount);
+	}
+
+	/**
+	 * Replay the binlog, if an error is encountered, the replay will be terminated,
+	 * and you need to confirm whether to skip the position to continue execution
+	 *
+	 * @param binlogFile binlog file
+	 */
+	private void replayBinlog(File binlogFile) {
+		if (!binlogFile.exists()) {
+			LOGGER.warn("File does not exist, {}", binlogFile.getAbsoluteFile());
+			return;
+		}
+
+		LOGGER.info("Start replay binlog file: {}", binlogFile.getAbsoluteFile());
+		EventDeserializer eventDeserializer = new EventDeserializer();
+		eventDeserializer.setCompatibilityMode(
+				EventDeserializer.CompatibilityMode.DATE_AND_TIME_AS_LONG_MICRO,
+				EventDeserializer.CompatibilityMode.CHAR_AND_BINARY_AS_BYTE_ARRAY,
+				EventDeserializer.CompatibilityMode.INVALID_DATE_AND_TIME_AS_MIN_VALUE
+		);
+
+		String position = null;
+		try (BinaryLogFileReader reader = new BinaryLogFileReader(binlogFile, eventDeserializer)) {
+			RowMap row = getRow(reader, binlogFile.getName());
+			while (row != null) {
+				if (shouldOutputRowMap(row)) {
+					producer.push(row);
+					rowCount++;
+				}
+				position = row.getPosition().getBinlogPosition().fullPosition();
+
+				// continue to get next
+				row = getRow(reader, binlogFile.getName());
+			}
+		} catch (Exception e) {
+			throw new RuntimeException("Replay failed, Check from: " + position + ", error: " + e.getMessage(), e);
+		} finally {
+			LOGGER.info("End replay binlog file: {}", binlogFile.getAbsoluteFile());
+		}
+	}
+
+	private RowMap getRow(BinaryLogFileReader reader, String binlogName) throws Exception {
```

Comment:
maybe we could find a way to refactor the original class where this came from to re-use it?  Like maybe turn it into a static method that takes `rowBuffer` as an argument or something?

Reply:
I added a BinlogConnectorEventProcessor class to handle the event - related data, removed the BinlogConnectorReplicator section method